libsolidity/formal/PredicateSort.cpp

```diff
@@ -41,7 +41,7 @@ SortPointer nondetInterfaceSort(ContractDefinition const& _contract, SymbolicSta
 	auto varSorts = stateSorts(_contract);
 	vector<SortPointer> stateSort{_state.stateSort()};
 	return make_shared<FunctionSort>(
-		stateSort + varSorts + stateSort + varSorts,
+		stateSort + varSorts + stateSort + varSorts + vector<SortPointer>{_state.errorFlagSort(), _state.thisAddressSort()},
```

Comment:
Can you also change the spec in `PredicateSort.h`?

Reply:
Updated.